compiler/rustc_mir_build/src/thir/pattern/check_match.rs

```diff
@@ -426,10 +428,18 @@ impl<'p, 'tcx> MatchVisitor<'_, 'p, 'tcx> {
                 let end_span = semi_span.shrink_to_lo();
                 let count = witnesses.len();
 
+                // If the pattern to match is an integer literal, perhaps the user meant to create a binding instead.
+                let int_suggestion = if let PatKind::Lit(expr) = &pat.kind && bindings.is_empty() {
+                    if let ExprKind::Lit(Spanned { node: LitKind::Int(_, _), span }) = expr.kind
+                    {
+                        Some(MiscPatternSuggestion::AttemptedIntegerLiteral { start_span: span.shrink_to_lo() })
+                    } else { None }
+                } else { None };
```

Comment:
You can use let chains here (which you're already leveraging on the first line) for reducing the number of branches in the code:

```suggestion
                let int_suggestion = if let PatKind::Lit(expr) = &pat.kind
                    && bindings.is_empty()
                    && let ExprKind::Lit(Spanned { node: LitKind::Int(_, _), span }) = expr.kind
                {
                        Some(MiscPatternSuggestion::AttemptedIntegerLiteral { start_span: span.shrink_to_lo() })
                } else {
                    None
                };
```